tests/infra/logging_app.py

```diff
@@ -2,60 +2,117 @@
 # Licensed under the Apache 2.0 License.
 
 import infra.checker
+import time
+import requests
 
 from loguru import logger as LOG
 
 
 class LoggingTxs:
-    def __init__(self, notifications_queue=None):
+    def __init__(
+        self,
+        notifications_queue=None,
+        tx_index_start=0,
+        can_fail=False,
+        wait_for_sync=True,
+        timeout=3,
+    ):
         self.pub = {}
         self.priv = {}
-        self.next_pub_index = 0
-        self.next_priv_index = 0
+        self.next_pub_index = tx_index_start
+        self.next_priv_index = tx_index_start
         self.notifications_queue = notifications_queue
+        self.can_fail = can_fail
```

Comment:
Should the `can_fail` be passed to `issue()` instead of being an attribute of the whole class?